functests/performance/updating_profile.go

```diff
@@ -204,43 +200,60 @@ var _ = Describe("[rfe_id:28761] Updating parameters in performance profile", fu
+			waitForMcpCondition(testutils.RoleWorker, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue, mcpUpdateTimeout)
 
-		// Check if node is Ready
-		for i := range node.Status.Conditions {
-			if node.Status.Conditions[i].Type == corev1.NodeReady {
-				Expect(node.Status.Conditions[i].Status).To(Equal(corev1.ConditionTrue))
+			// Check if node is Ready
+			for i := range newCnfNode.Status.Conditions {
+				if newCnfNode.Status.Conditions[i].Type == corev1.NodeReady {
+					Expect(newCnfNode.Status.Conditions[i].Status).To(Equal(corev1.ConditionTrue))
+				}
 			}
-		}
 
-		// check that the pre-boot-tuning script and service removed
-		out, _ := nodes.ExecCommandOnMachineConfigDaemon(testclient.Client, &node, []string{"ls", "/rootfs/" + testutils.PerfRtKernelPrebootTuningScript})
-		Expect(out).To(ContainSubstring("No such file or directory"))
-		out, _ = nodes.ExecCommandOnMachineConfigDaemon(testclient.Client, &node, []string{"ls", "/rootfs/" + "/etc/systemd/system/pre-boot-tuning.service"})
-		Expect(out).To(ContainSubstring("No such file or directory"))
+			// check that the pre-boot-tuning script and service removed
+			out, _ := nodes.ExecCommandOnMachineConfigDaemon(testclient.Client, newCnfNode, []string{"ls", "/rootfs/" + testutils.PerfRtKernelPrebootTuningScript})
+			Expect(out).To(ContainSubstring("No such file or directory"))
+			out, _ = nodes.ExecCommandOnMachineConfigDaemon(testclient.Client, newCnfNode, []string{"ls", "/rootfs/" + "/etc/systemd/system/pre-boot-tuning.service"})
+			Expect(out).To(ContainSubstring("No such file or directory"))
+
+			// check that the configs reverted
+			Expect(execCommandOnWorker(chkKernel, newCnfNode)).NotTo(ContainSubstring("PREEMPT RT"))
+			Expect(execCommandOnWorker(chkCmdLine, newCnfNode)).NotTo(ContainSubstring("tuned.non_isolcpus"))
+			Expect(execCommandOnWorker(chkKubeletConfig, newCnfNode)).NotTo(ContainSubstring("reservedSystemCPUs"))
+		})
 
-		// check that the configs reverted
-		Expect(execCommandOnWorker(chkKernel, &node)).NotTo(ContainSubstring("PREEMPT RT"))
-		Expect(execCommandOnWorker(chkCmdLine, &node)).NotTo(ContainSubstring("tuned.non_isolcpus"))
-		Expect(execCommandOnWorker(chkKubeletConfig, &node)).NotTo(ContainSubstring("reservedSystemCPUs"))
+		It("Reverts back nodeSelector and cleaning up leftovers", func() {
+			nodeSelector := fmt.Sprintf(`"%s/%s": ""`, testutils.LabelRole, testutils.RoleWorkerRT)
+			Expect(testclient.Client.Patch(context.TODO(), profile,
+				client.ConstantPatch(
+					types.JSONPatchType,
+					[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec/nodeSelector", "value": {%s} }]`, nodeSelector)),
+				),
+			)).ToNot(HaveOccurred())
+			Expect(testclient.Client.Delete(context.TODO(), mcp)).ToNot(HaveOccurred())
+			waitForMcpCondition(testutils.RoleWorkerRT, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue, timeout)
```

Comment:
don't we need to wait for `Updating` first?
maybe even factor this out, it is used at several places:
1. wait for updating
2. wait for updated
3. check node readiness (not sure if this needed at all? when a node isn't ready, the MCP can't be not updated, can it?)

Reply:
I'm not sure we really need this waitForMcpCondition() here at all..

As I saw, when I changed nodeSelector in profile - worker-rt node (that initially had been updated) was still remained updated. So when I'm returning nodeSelector back - mcp does not start any updates since the node is already updated.
  
Actually I'm not sure may be it is not expected behavior that node is keeping updated after changing nodeSelector and it should be reverted back to **worker** state.. (or may be it takes some time for operator to realize that the node should be reverted back)

Reply:
> when I changed nodeSelector in profile - worker-rt node (that initially had been updated) was still remained updated

Oh... I think that would be a bug then?
Should we check in `[test_id:28440]Verifies that nodeSelector can be updated in performance profile` that the original mcp updates and reverts it's nodes to normal worker...? :thinking:

Reply:
Verified manually: It took more than 10 minutes for MCP to start reverting back worker-rt node after changing NodeSelector from worker-rt to worker-test..

Reply:
wow
good to know it does (and we should test it)
but strange that it is so slow
thx for testing :+1:

Reply:
looks like it was enough time for completing tests before returning nodeSelector to initial state.. that's why i didn't have additinal MCP updates

Reply:
not sure if we miss something here... but it can be handled in a follow up if needed

/lgtm

Reply:
Thanks, I'll keep an eye on it.